.github/workflows/docs.yml

```diff
@@ -0,0 +1,62 @@
+name: CI
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push events but only for the latest branch
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '**'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Runs a set of commands using the runners shell
+      - name: Run a multi-line script
+        run: |
+          echo "Install Sphinx and deploy docs..."
+          cd rust
+          cargo build
+          cd ..
+          cd docs
+          python --version
+          python -m venv venv
+          source venv/bin/activate
+          echo "Upgrade pip and install requirements."
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          echo "Create docs."
+          make versions
+          cp -r build /tmp
+          cd ..
+          git fetch
+          git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          git config --global user.name "$GITHUB_ACTOR"
+          git checkout gh-pages
+          rm -rf *
+          # TODO: Update to final hostname.
+          echo docz.opendp.org > CNAME
+          echo "for underscore directories" > .nojekyll
+          cp -r /tmp/build/html/* .
+          git add --all --force
+      - name: Push docs to gh-pages branch
+        if: success() && github.ref == 'refs/heads/main'
```

Comment:
How often do we want to update the documentation? Per-merge or per-release? 
In this PR https://github.com/opendp/opendp/pull/162, CI will release a new python package and crate whenever there is a push to the `release` branch. If you tie documentation deployment to the release branch, then the documentation will update less frequently, only when a release is cut.

Reply:
I guess it could update the docs each merge into main, where the updated docs includes both the `latest` branch (renamed from `main`) and all of the tags. 

Would checking out from main to a new branch `latest` in the ci script not work?

Reply:
At least for the time being, while docs are being worked on quite actively, I was thinking the docs site should be updated when pull requests were merged.

In this pull request, the updated docs would appear under "main" (docs.opendp.org/en/main), reflecting the fact that the pull request was merged into the main branch.

Assuming a regular release cycle in the future, I'm open to only updating docs on release. That's what we do for Dataverse. That said, even in the context of Dataverse, when we haven't released for a while, I'm often interested in seeing what the updated docs will look like, so I think there's value in always having "main" or "latest" to be able to preview what's coming in the next release.

> I guess it could update the docs each merge into main, where the updated docs includes both the `latest` branch (renamed from `main`) and all of the tags.

I'm not sure I fully grok this. Is renaming `main` on the table? The extension I'm using does support tags. No problem there. That's my plan for people being able to switch between different versions of the library.

I don't think the `checkout` above will work but in the same vein, we might be able to have the CI whatever is in `main` to a branch called `latest` and then run `make versions`. I haven't tried this yet.

Reply:
Right, if we don't rename main, then each time main is pushed to, CI could run something like:
```
git branch -D latest && git checkout -b latest
make versions
```
By "rename from main" I was referring to the checkout. I don't think I'm the guy to decide if we will rename main. I agree that it makes sense to push docs more frequently to a `latest` version.

Reply:
Oh, I see. Delete the `latest` branch and then create it again from `main`. Yeah, I would hope that would work. I was thinking of a merge from `main` to `latest` but I'm not picky. 😄

Reply:
I'm just brainstorming. `merges` would work too.